src/cephadm/cephadm.py

```diff
@@ -2579,7 +2808,7 @@ def deploy_daemon(
     with write_new(data_dir + '/unit.configured', owner=(uid, gid)) as f:
         f.write('mtime is time we were last configured\n')
 
-    update_firewalld(ctx, daemon_type)
+    update_firewalld(ctx, daemon_form_create(ctx, ident))
```

Comment:
```
2023-10-03T20:22:10.202 INFO:journalctl@ceph.mgr.a.smithi131.stdout:Oct 03 20:22:10 smithi131 conmon[100955]: Redeploy daemon agent.smithi131 ...
2023-10-03T20:22:10.202 INFO:journalctl@ceph.mgr.a.smithi131.stdout:Oct 03 20:22:10 smithi131 conmon[100955]: Traceback (most recent call last):
2023-10-03T20:22:10.202 INFO:journalctl@ceph.mgr.a.smithi131.stdout:Oct 03 20:22:10 smithi131 conmon[100955]:   File "/usr/lib64/python3.6/runpy.py", line 193, in _run_module_as_main
2023-10-03T20:22:10.202 INFO:journalctl@ceph.mgr.a.smithi131.stdout:Oct 03 20:22:10 smithi131 conmon[100955]:     "__main__", mod_spec)
2023-10-03T20:22:10.202 INFO:journalctl@ceph.mgr.a.smithi131.stdout:Oct 03 20:22:10 smithi131 conmon[100955]:   File "/usr/lib64/python3.6/runpy.py", line 85, in _run_code
2023-10-03T20:22:10.202 INFO:journalctl@ceph.mgr.a.smithi131.stdout:Oct 03 20:22:10 smithi131 conmon[100955]:     exec(code, run_globals)
2023-10-03T20:22:10.202 INFO:journalctl@ceph.mgr.a.smithi131.stdout:Oct 03 20:22:10 smithi131 conmon[100955]:   File "/tmp/tmpu1dubqmz.cephadm.build/__main__.py", line 7642, in <module>
2023-10-03T20:22:10.202 INFO:journalctl@ceph.mgr.a.smithi131.stdout:Oct 03 20:22:10 smithi131 conmon[100955]:   File "/tmp/tmpu1dubqmz.cephadm.build/__main__.py", line 7630, in main
2023-10-03T20:22:10.203 INFO:journalctl@ceph.mgr.a.smithi131.stdout:Oct 03 20:22:10 smithi131 conmon[100955]:   File "/tmp/tmpu1dubqmz.cephadm.build/__main__.py", line 5080, in command_deploy_from
2023-10-03T20:22:10.203 INFO:journalctl@ceph.mgr.a.smithi131.stdout:Oct 03 20:22:10 smithi131 conmon[100955]:   File "/tmp/tmpu1dubqmz.cephadm.build/__main__.py", line 5098, in _common_deploy
2023-10-03T20:22:10.203 INFO:journalctl@ceph.mgr.a.smithi131.stdout:Oct 03 20:22:10 smithi131 conmon[100955]:   File "/tmp/tmpu1dubqmz.cephadm.build/__main__.py", line 5247, in _dispatch_deploy
2023-10-03T20:22:10.203 INFO:journalctl@ceph.mgr.a.smithi131.stdout:Oct 03 20:22:10 smithi131 conmon[100955]:   File "/tmp/tmpu1dubqmz.cephadm.build/__main__.py", line 2811, in deploy_daemon
2023-10-03T20:22:10.203 INFO:journalctl@ceph.mgr.a.smithi131.stdout:Oct 03 20:22:10 smithi131 conmon[100955]:   File "/tmp/tmpu1dubqmz.cephadm.build/cephadmlib/daemon_form.py", line 124, in create
2023-10-03T20:22:10.203 INFO:journalctl@ceph.mgr.a.smithi131.stdout:Oct 03 20:22:10 smithi131 conmon[100955]:   File "/tmp/tmpu1dubqmz.cephadm.build/cephadmlib/daemon_form.py", line 120, in choose
2023-10-03T20:22:10.203 INFO:journalctl@ceph.mgr.a.smithi131.stdout:Oct 03 20:22:10 smithi131 conmon[100955]: cephadmlib.daemon_form.UnexpectedDaemonTypeError: 'agent'
```
It seems that this has broken deployment of the agent.

Reply:
https://pulpito.ceph.com/adking-2023-10-03_19:55:53-orch:cephadm-wip-adk-testing-2023-10-03-1223-distro-default-smithi/7409863/
https://pulpito.ceph.com/adking-2023-10-03_19:55:53-orch:cephadm-wip-adk-testing-2023-10-03-1223-distro-default-smithi/7409891